File: Codility/003_03_TapeEquilibrium.cpp
// you can use includes, for example:
// #include <algorithm>
#include <vector>
#include <algorithm>
// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)
    int gap = 0;
    int tempGap = 0;
    int N = A.size();

    int left = A[0];
    int right = A[1];
    for(int i = 2; i < N; i++) {
        right += A[i];
    }

    gap = abs(left-right);
    for(int i = 1; i < N-1; i++) {
        left += A[i];
        right -= A[i];
        tempGap = abs(left-right);
        if(tempGap < gap)
            gap = tempGap;
    }


    return gap;
}

File: Codility/003_03_TapeEquilibrium.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        int N = A.length;
        int ans = 0;

        // P1
        int left = A[0];
        int right = A[1];
        for(int i = 2; i < N; i++) {
            right += A[i];
        }

        // P2~
        int gap = Math.abs(left-right);
        for(int i = 1; i < N-1; i++) {
            left += A[i];
            right -= A[i];
            int tempGap = Math.abs(left-right);
            if(gap > tempGap)
                gap = tempGap;
        }

        return gap;
    }
}
